### PR TITLE
improve: improve typing in src/clients/bridges/

### PR DESCRIPTION
**Pull request by mrice32**
This is the first PR in what will hopefully be a series of PRs to address type issues that are blocking us from enabling the `strict` setting in our tsconfig.

The idea behind this change is to make our bots safer and more error-proof by expanding the sort of type-checking we're getting from typescript.

I've created this by enabling strict locally and working through the errors manually. Since there are hundreds of errors, I've decided to start working through them folder-by-folder and doing PRs as I go so these can be carefully reviewed.

### CODE REVIEW

File: src/clients/bridges/PolygonAdapter.ts [hunk 1]
```diff
@@ -110,14 +123,17 @@ export class PolygonAdapter extends BaseAdapter {
 
         const l2Token = this.getL2Token(l1Token);
 
+        if (l2Token == null) continue;
```
Comment (kevinuma) [inline src/clients/bridges/PolygonAdapter.ts:126]: nit: l2Token === null

**Review by kevinuma — APPROVED** [review]

File: src/clients/bridges/PolygonAdapter.ts (hunk 1 above)
Comment (mrice32) [reply src/clients/bridges/PolygonAdapter.ts:126]: Nice catch!

Comment (mrice32) [reply src/clients/bridges/PolygonAdapter.ts]: Fixed!

**Review by kevinuma — APPROVED** [review]

File: src/clients/bridges/PolygonAdapter.ts [hunk 2]
```diff
@@ -182,46 +201,52 @@ export class PolygonAdapter extends BaseAdapter {
...
-        this.log("Could not construct l1Bridge. Likely misconfiguration", { l1Token, error, tokenToBridge }, "error");
-        return null;
-      }
+    else return new Contract(l1RootChainManager, polygonL1RootChainManagerInterface, this.getSigner(1));
   }
 
   // Note that on polygon we dont query events on the L2 bridge. rather, we look for mint events on the L2 token.
   getL2Token(l1Token: string): Contract | null {
-    try {
+    if (this.isSupportedToken(l1Token))
```
Comment (nicholaspai) [inline src/clients/bridges/PolygonAdapter.ts:233]: Doesn't converting from a try-catch to an if-else change behavior?

File: src/utils/TypeGuards.ts [hunk 3]
```diff
@@ -9,3 +9,7 @@ export function isPromiseRejected<T>(
 ): promiseSettledResult is PromiseRejectedResult {
   return promiseSettledResult.status === "rejected";
 }
+
+export function isDefined<T>(input: T | null | undefined): input is T {
```
Comment (nicholaspai) [inline src/utils/TypeGuards.ts:13]: +1

**Review by nicholaspai — APPROVED** [review]
LGTM

File: src/clients/bridges/PolygonAdapter.ts (hunk 2 above)
Comment (mrice32) [reply src/clients/bridges/PolygonAdapter.ts:233]: Yep, I think this is a behavior change.

Based on the error message, I assumed all we were catching was an unsupported token, though? Or are there other cases we were trying to catch here?

**Pull request merged**